### PR TITLE
switch network for balance

### PR DESCRIPTION
**Pull request by ttaoS**

### CODE REVIEW

File: src/ts/components/dashboard/TopMenu.tsx [hunk 1]
```diff
@@ -38,7 +52,28 @@ class TopMenu extends React.Component<ITopMenuProps, ITopMenuState> {
     if (this.props.settings.selectedAccount) {
       this.props.getTransactions(this.props.settings.selectedAccount.address, data.value)
     }
+    const { apiContext } = this.props
+    this.setState({ isLoading: true })
+    apiContext.provider && this.props.destroyApi(apiContext.provider)
+  }
 
+  recreateApi = () => {
```
Comment (ttaoS) [inline src/ts/components/dashboard/TopMenu.tsx]: we probably can refactor this one to a common util later.

File: src/ts/components/dashboard/TopMenu.tsx [hunk 2]
```diff
@@ -1,33 +1,47 @@
...
 import { withRouter, RouteComponentProps } from 'react-router'
-import { Image, Grid } from 'semantic-ui-react'
+import { Image, Grid, Dimmer, Loader } from 'semantic-ui-react'
 import { networks } from '../../constants/networks'
+import { WsProvider } from '@polkadot/rpc-provider'
 import { IAppState } from '../../background/store/all'
 import { saveSettings } from '../../background/store/settings'
 import { getTransactions } from '../../background/store/transaction'
-import { ChainDropdown } from '../basic-components'
+import {ChainDropdown} from '../basic-components'
```
Comment (mytechtip) [inline src/ts/components/dashboard/TopMenu.tsx]: minor style inconsistency if want to change

File: src/ts/components/dashboard/TopMenu.tsx [hunk 3]
```diff
@@ -1,33 +1,47 @@
...
 import styled from 'styled-components'
 import SettingsMenu from './SettingsMenu'
+import { ApiOptions } from "@polkadot/api/types";
+import { createApi, destroyApi } from "../../background/store/api-context";
 
 interface ITopMenuProps extends StateProps, DispatchProps, RouteComponentProps {}
 
 interface ITopMenuState {
+  isLoading: boolean,
```
Comment (fyang1024) [inline src/ts/components/dashboard/TopMenu.tsx]: Can we remove this flag and instead use apiContext.apiReady as Dimer's toggling flag? This way user does not have to wait at least 3 seconds?

File: src/ts/components/dashboard/TopMenu.tsx [hunk 4]
```diff
@@ -38,7 +52,28 @@ class TopMenu extends React.Component<ITopMenuProps, ITopMenuState> {
...
 
+  recreateApi = () => {
+    const { apiContext, settings } = this.props
+    if (!apiContext.apiReady && settings) {
+      this.setState({ ...this.state, tries: this.state.tries++ })
+      const network = networks[settings.network]
+      const provider = new WsProvider(network.rpcServer)
+      let apiOptions: ApiOptions = { provider, types: network.types }
+      this.props.createApi(apiOptions)
+      if (this.state.tries <= 5) {
```
Comment (fyang1024) [inline src/ts/components/dashboard/TopMenu.tsx]: do we need else as network error handling?

File: src/ts/components/dashboard/TopMenu.tsx [hunk 5]
```diff
@@ -38,7 +52,28 @@ class TopMenu extends React.Component<ITopMenuProps, ITopMenuState> {
...
       this.props.getTransactions(this.props.settings.selectedAccount.address, data.value)
     }
+    const { apiContext } = this.props
+    this.setState({ isLoading: true })
+    apiContext.provider && this.props.destroyApi(apiContext.provider)
+  }
 
+  recreateApi = () => {
+    const { apiContext, settings } = this.props
+    if (!apiContext.apiReady && settings) {
```
Comment (fyang1024) [inline src/ts/components/dashboard/TopMenu.tsx]: why do we need check `settings` here? If we don't have that, we can do 
```
if (apiContext.apiRead) {
} else {
}
```
which seems more readable

File: src/ts/components/dashboard/TopMenu.tsx (hunk 3 above)
Comment (ttaoS) [reply src/ts/components/dashboard/TopMenu.tsx]: it still need 3 sec regardless. but we probably can save a flag.

Comment (fyang1024) [reply src/ts/components/dashboard/TopMenu.tsx]: The `isLoading` can be set to `false` in at least 3 seconds when the timeout happens, while `apiContext.apiReady` is set to `true` by `redux` as soon as the api is created, correct me if I am wrong.

Comment (ttaoS) [reply src/ts/components/dashboard/TopMenu.tsx]: I see. you are right.

Comment (ttaoS) [reply src/ts/components/dashboard/TopMenu.tsx]: updated

**Pull request merged**